students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part2.py

```diff
@@ -0,0 +1,49 @@
+# Factorial
+n = int(input())
+silnia = 1
+for x in range (1, n+1) :
+    silnia *= x
+print (silnia)
+
+
+# The number of zeros
+n = int(input())
+zeros_number = 0
+for x in range (n) :
+    if int(input()) == 0 :
+        zeros_number += 1
+print (zeros_number)
+
+
+# Adding factorials
+n = int(input())
+summary = 0
+silnia = 1
+for x in range (1, n+1) :
+    silnia *= x
+    summary += silnia
+print (summary)
+
+
+# Ladder
+n = int(input())
+for x in range (1, n+1):
+    for y in range (1, x+1):
+        print (y, sep='', end='')
+    print ('')
+
+
+# Lost card
+n = int(input())
+cards_list = []
+for x in range (n-1):
+    cards_list.append(int(input()))
+for x in range (1, n+1) :
```

Comment:
the algorithm here could be simplified. We should avoid nesting loops if not necessary.
Please check and analyze solution below:
```
N = int(input())
all_cards = list(range(1,N+1))
for i in range(1,N):
    all_cards.remove(int(input()))
print(all_cards[0])      
```